File: src/Web/Models/Search/OrderSearchModel.cs
using Core.Entities;
using Microsoft.AspNetCore.Mvc;

namespace Web.Models
{
    public class OrderSearchModel
    {
        [FromQuery(Name = "id")]
        public int[] Ids { get; set; }
        [FromQuery(Name = "comment")]
        public string[] Comments { get; set; }
        [FromQuery(Name = "status")]
        public DeliveryStatus[] Statuses { get; set; }
        [FromQuery(Name = "buyerId")]
        public int[] BuyerIds { get; set; }
        [FromQuery(Name = "userOrAnonymousId")]
        public string[] UserOrAnonymousIds { get; set; }
        [FromQuery(Name = "addressId")]
        public int[] AddressIds { get; set; }
    }
}

File: src/Web/ClientApp/src/models/search/OrderSearchModel.ts
import {DeliveryStatus} from '../order/Order';

export class OrderSearchModel {
  constructor(
    public ids?: number[],
    public comments?: string[],
    public statuses?: DeliveryStatus[],
    public buyerIds?: number[],
    public addressIds?: number[],
    public userOrAnonymousIds?: string[],
  ) {
  }
}
